src/uct/sm/scopy/base/scopy_iface.c

```diff
@@ -130,6 +135,11 @@ unsigned uct_scopy_iface_progress(uct_iface_h tl_iface)
     unsigned count           = 0;
 
     ucs_arbiter_dispatch(&iface->arbiter, 1, uct_scopy_ep_progress_tx, &count);
+
+    if (ucs_unlikely(ucs_arbiter_is_empty(&iface->arbiter))) {
+        uct_scopy_iface_disable_progress(iface);
+    }
```

Comment:
i think that in case of scopy, iface_progress / iface_progress_enable / iface_progress_disable should do nothing, and scopy should add/remove the callback to/from worker callback queue when needed

Reply:
@yosefe good catch!
done, changes were force-pushed, since the implementation was changed, the old one is not relevant anymore